### PR TITLE
Fixes #33758 - update foreman-js 8.16.0

### PR DESCRIPTION
**Pull request by MariaAga**
packaging pr:https://github.com/theforeman/foreman-packaging/pull/7188
new pf4 has flyouts navigation, will open a pr for it soon :)
Also has a small fix for thats needed for REX select components.
Looks like the jest revert worked as Katello tests are running ok except for 3 snapshot failures

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,19 +21,19 @@
...
-    "@theforeman/builder": ">= 8.7, < 8.12.1",
-    "@theforeman/eslint-plugin-foreman": ">= 8.7, < 8.12.1",
-    "@theforeman/stories": ">= 8.7, < 8.12.1",
-    "@theforeman/test": ">= 8.7, < 8.12.1",
-    "@theforeman/vendor-dev": ">= 8.7, < 8.12.1",
+    "@theforeman/builder": "^8.16.0",
+    "@theforeman/eslint-plugin-foreman": "^8.16.0",
+    "@theforeman/stories": "^8.16.0",
+    "@theforeman/test": "^8.16.0",
+    "@theforeman/vendor-dev": "^8.16.0",
```
Comment (Ron-Lavi) [inline package.json:34]: can you also apply changes from https://github.com/theforeman/foreman/pull/8813/files
react-intl is now part of @theforeman/vendor and eslint plugin can move to devDep after the gh action issue is resolved.

Comment (MariaAga) [reply package.json:34]: Yeah, added

**Review by Ron-Lavi — COMMENTED** [review]
Three Katello snapshots are failing, non-blocking for this PR
but needs an update in Katello
cc @sjha4 @jeremylenz @Andrewgdewar

**Review by Ron-Lavi — APPROVED** [review]
Thanks @MariaAga

**Comment by sharvit** [comment]
https://github.com/Katello/katello/pull/9739 should fix the katello errors

**Review by tbrisker — APPROVED** [review]
Thanks @MariaAga and @laviro !

**Pull request merged**